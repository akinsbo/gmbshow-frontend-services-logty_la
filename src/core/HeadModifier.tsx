/**
 * This class 
 */
import * as React from "react"
// tslint:disable-next-line:import-name
import { Helmet } from "react-helmet-async"

export interface HeadModifierProps {
    title: string,
    author: string,
    link: string,
    theme_color: string
}

// 'AppRouterProps' describes the shape of props.
// State is never set so we use the '{}' type.
export default class HeadModifier extends React.Component<HeadModifierProps, {}> {

    static defaultProps = {
        author: "Olaolu Akinsete",
        link: "https://duyono.com",
        theme_color: "#008f68",
        title: "duyono"
    }

    render() {

        const { title, link } = this.props
        return (
            // Links would be in another file
            <div>
                <Helmet titleTemplate={`%s | ${title}`}>
                    <title>{title}</title>
                    <meta name="link" content={link} />
                    {/* <meta name="theme-color" content={this.props.siteData.theme_color} /> */}
                </Helmet>
            </div >
        )
    }
}
