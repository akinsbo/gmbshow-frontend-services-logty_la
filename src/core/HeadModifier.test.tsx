import * as Enzyme from "enzyme"
import enzymeAdapterReact16 from "enzyme-adapter-react-16"
import "jest-enzyme"
import * as React from "react"
import { HelmetProvider } from "react-helmet-async"
import * as renderer from "react-test-renderer"
import HeadModifier from "./HeadModifier"


Enzyme.configure({
    adapter: new enzymeAdapterReact16()
})

const siteDataJson = {
    author: "Olaolu Akinsete",
    link: "https://duyono.com",
    theme_color: "#008f68",
    title: "duyono"
}


describe("HeadModifier", () => {
    it("should render", () => {
        const renderWrapper = renderer.create(
            <HelmetProvider>
                <HeadModifier {...siteDataJson} />
            </HelmetProvider>
        )
        expect(renderWrapper).toMatchSnapshot()
    })
    describe("structure", () => {
        const wrapper = Enzyme.shallow(
            <HeadModifier {...siteDataJson} />
        )
        describe("Helmet", () => {
            let helmetWrapper: any
            beforeEach(() => {
                helmetWrapper = wrapper.find("Helmet")
            })
            it("should be present", () => {
                expect(helmetWrapper).toHaveLength(1)
            })
            it("should contain title element", () => {
                expect(helmetWrapper.find("title"))
                    .toHaveLength(1)
            })
            describe("meta", () => {
                let metaWrapper: any
                beforeEach(() => {
                    metaWrapper = helmetWrapper.find("meta")
                })
                it("should be present in Helmet", () => {
                    expect(metaWrapper).toHaveLength(1)
                })
                it("should contain props", () => {
                    expect(metaWrapper).toHaveProp("name")
                })
            })// describe meta
        })// describe Helmet
    })// describe strucutre
})// describe HeadModifier


