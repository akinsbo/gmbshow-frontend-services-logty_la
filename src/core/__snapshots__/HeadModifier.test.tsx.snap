// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadModifier should render 1`] = `<div />`;
